### PR TITLE
Update sync endpoint

### LINKED ISSUE #91: Update sync endpoint
The following modifications to the `GET /sync` endpoint  need to be made:
  - Add `online` and `unavailable` variants to `SetPresence` (0.4.0/matrix-org/matrix-doc#1141)
  - Add `device_one_time_keys_count` to response (0.4.0/matrix-org/matrix-doc#1284)
  - Add `to_device` parameter

There may be other changes necessary, so so the [spec documentation](https://matrix.org/docs/spec/client_server/r0.6.0#get-matrix-client-r0-sync) for more information.

### PR DESCRIPTION
**Pull request by iinuwa**
Fixes #91.

### CODE REVIEW

**Comment by iinuwa** [comment]
@jplatte There are quite a few fields that are not marked as required in the spec, but are also not `Option` types. Should those all be modified?

**Comment by jplatte** [comment]
If they don't have `#[serde(default)]`, then yeah, they should either be `Option`s, or have a `#[serde(deafult)]` (depending on whether there's a semantic distinction between the default value and no value).

File: src/r0/sync/sync_events.rs [hunk 1]
```diff
@@ -64,15 +66,27 @@ ruma_api! {
...
         #[wrap_incoming]
         pub to_device: ToDevice,
+        /// Information on E2E device updates.
+        /// Only present on an incremental sync.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub device_lists: Option<DeviceLists>,
+        /// For each key algorithm, the number of unclaimed one-time keys
+        /// currently held on the server for a device.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub device_one_time_keys_count: Option<HashMap<KeyAlgorithm, UInt>>
```
Comment (jplatte) [inline src/r0/sync/sync_events.rs]:
```suggestion
        #[serde(skip_serializing_if = "HashMap::is_empty")]
        pub device_one_time_keys_count: HashMap<KeyAlgorithm, UInt>,
```

File: src/r0/sync/sync_events.rs [hunk 2]
```diff
@@ -186,10 +206,12 @@ pub struct UnreadNotificationsCount {
 #[derive(Clone, Debug, Serialize, Outgoing)]
 pub struct Timeline {
     /// True if the number of events returned was limited by the `limit` on the filter.
-    pub limited: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub limited: Option<bool>,
```
Comment (jplatte) [inline src/r0/sync/sync_events.rs]: Do the three possibilities here (field absent, `true`, `false`) have three distinct meanings or does the field being absent mean the same as a value of `true` or `false`?

Comment (iinuwa) [reply src/r0/sync/sync_events.rs]: Ah, no. I added that, thinking to save a few bytes on serialization, but it doesn't have a semantic difference.

**Comment by iinuwa** [comment]
If the types don't implement `Default`, then should I create an implementation or use `Option`?

**Comment by jplatte** [comment]
Which types are we talking about? If you create a `Default` implementation, you also need to create an `is_something` method to use with skip_serializing_if. If the "something" in that is hard to find a concrete name for (other than "default") I would probably keep the `Option` wrapping. I hope you know what I mean / that wasn't too cryptic 😅

**Pull request merged**